src/examples/ProofOfExistence.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState, useEffect } from "react";
+import { Form, Input, Grid, Message } from "semantic-ui-react";
+import { blake2AsHex } from "@polkadot/util-crypto";
+
+import TxButton from "../TxButton";
+
+export default function ProofOfExistence(props) {
+  const { api, accountPair } = props;
+  const [status, setStatus] = useState("");
+  const [digest, setDigest] = useState("");
+  const [owner, setOwner] = useState("");
+  const [block, setBlock] = useState(0);
+
+  let fileReader;
+
+  const bufferToDigest = () => {
+    const content = Array.from(new Uint8Array(fileReader.result))
+      .map(b => b.toString(16).padStart(2, "0"))
+      .join("");
+
+    const hash = blake2AsHex(content, 256);
+    setDigest(hash);
+  };
+
+  const handleFileChosen = file => {
+    fileReader = new FileReader();
+    fileReader.onloadend = bufferToDigest;
+    fileReader.readAsArrayBuffer(file);
+  };
+
+  useEffect(() => {
+	let unsubscribe;
+
+	api.query.poe.proofs(digest, (result) => {
+		setOwner(result[0].toString());
+		setBlock(result[1].toNumber());
+	}).then(unsub => {
+		unsubscribe = unsub;
+	});
+
+	return () => unsubscribe && unsubscribe();
+  }, [digest, api.query.poe]);
+
+  return (
+    <Grid.Column>
+      <h1>Proof Of Existence</h1>
+      <Form success={!!digest && block === 0} warning={block !== 0}>
+        <Form.Field>
+          <Input
+            type="file"
+            id="file"
+            label="Your File"
+            onChange={e => handleFileChosen(e.target.files[0])}
+          />
+          <Message success header="File Digest Unclaimed" content={digest} />
+          <Message
+            warning
+            header="File Digest Claimed"
+            list={[digest,`Owner: ${owner}`,`Block: ${block}`]}
+          />
+        </Form.Field>
+
+        <Form.Field>
+          <TxButton
+            api={api}
+            accountPair={accountPair}
+            label={"Submit Proof"}
+            setStatus={setStatus}
+            params={[digest]}
+            tx={api.tx.poe.createClaim}
+          />
+          {status}
+        </Form.Field>
+      </Form>
```

Comment:
Should we also have a button to revoke a claim since there is a dispatchable call for that in the module?

Reply:
yeah, that is simple and totally makes sense